external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -616,6 +726,17 @@ fn module(
         context.restore_unscoped(unscoped.clone());
         constant(context, name, c)
     });
+    // Silence unused use fun warnings if a module has macros.
+    // For public macros, the macro will pull in the use fun, and we will which case we will be
+    //   unable to tell if it is used or not
+    //   TODO we could approximate this by just checking for the name, regardless of the type
+    // For private macros, we duplicate the scope of the module and when resolving the method
+    //   fail to mark the outer scope as used (instead we only mark the modules scope cloned
+    //   into the macro)
+    let has_macro = functions.iter().any(|(_, _, f)| f.macro_.is_some());
+    if has_macro {
+        mark_all_use_funs_as_used(&mut use_funs);
```

Comment:
Could we reduce this surface area to only the use funs the macro _could_ mention? that is, process the macro body and identify any possible method names and mark those, but leave the rest?

(That can be a follow-up diff.)

Reply:
Yeah agreed, but its a bit of work. We would have to walk the macro to find anything that is `.foo(` and mark `foo` as being unused. I'll highlight that more clearly